Lib/test/test_pyexpat.py

```diff
@@ -821,5 +823,169 @@ def start_element(name, _):
         self.assertEqual(started, ['doc'])
 
 
+class AttackProtectionTestCases:
+    """Generic interface for testing XML Expat protections.
+
+    The protections being tested should mitigate attacks based
+    on Billion Laughs payloads.
```

Comment:
```suggestion
    """Base class for testing protections against XML payloads with disproportionate amplification.

    The protections being tested should detect and prevent attacks that
    leverage disproportionate amplification from small input size.
    
    NOTE: The class does not derive from unittest.TestCase by itself so that the `test_*`
          methods in this parent class are not picked up by test discovery.
```

- because the allocation tracker is not specific to billion laughs by any means.
- the hiding of tests is a trick that seems worth documenting as non-accidental

Reply:
The trick is not something we document in CPython because it is common in other parts of the codebase.

What I meant by "billion laughs based" is that we check protections by using such payload but I will amend the description.